src/orng.cpp

```diff
@@ -413,61 +367,77 @@ void orng::requestrand(uint64_t assoc_id, uint64_t signing_value, const eosio::n
 }
 
 /* setrand - completes request */
-void orng::setrand(name oracle, uint64_t id, uint8_t ver, std::string sig){
-    eosio::check(!is_paused(), "paused");
-    require_auth(oracle);
-    uint64_t strikes_max = get_config(strikes_max_index, 0);
+ACTION orng::setrand(name oracle, uint64_t id, uint8_t ver, std::string sig){
+    checksum256 rnd = _validate_and_compute_rnd(oracle, id, ver, sig);
+    if (rnd == checksum256{}) return; // validation failed, oracle got strike
+
     uint64_t fee_per_call = get_config(fee_per_call_index, 0);
+    auto rit = req_table.require_find(id, "no request found");
 
-    auto oit = oracles_table.require_find(oracle.value, "unknown oracle");
-    check(!oit->suspended, "oracle suspended");
+    // Attempt direct delivery via inline action
+    action{
+        permission_level{get_self(), "active"_n},
+        rit->dapp, "receiverand"_n,
+        std::make_tuple(rit->assoc_id, rnd)
+    }.send();
 
-    auto rit = req_table.require_find(id, "no request found"); 
-    check(rit->ver == ver, "version mismatch");
+    // If we reach here, delivery succeeded - clean up request
+    req_table.erase(rit);
 
-    if(rit->status != 0) return; // already being processed
+    _reward_oracles(asset{static_cast<int64_t>(fee_per_call), WAX});
 
-    auto pit = pkey_table.require_find(ver, "key not found");
-    check(pit->retired == false, "key retired");
+    dec_job_count(rit->dapp);
```

Comment:
Do we still need count number of jobs for each Dapp? It was previously used to check max queue size per Dapp but I think we can remove it since we already manage that with `credits` and `fee_balance`.